### PR TITLE
Allow limits and offsets to use value args (? in prepared statements)

### LINKED ISSUE #439: Error: Unsupported feature: LIMIT with non-integer literal
Getting the following error:

```
Error 1105: unknown error: unsupported feature: LIMIT with non-integer literal
```

Even though the query I'm running actually does use an integer, it's just interpolated.

Minimal repo:
```go
_, err := db.Query(`SELECT * FROM foo LIMIT ?`, 10)
// err is "Error 1105: unknown error: unsupported feature: LIMIT with non-integer literal"
```

(maybe the error means non-literal integers are not supported? using `LIMIT 10` instead of interpolating works.)

### PR DESCRIPTION
**Pull request by zachmu**
This fixes #439

### CODE REVIEW

File: enginetest/queries.go [hunk 1]
```diff
@@ -1221,10 +1221,29 @@ var QueryTests = []QueryTest{
...
 	},
+	{
+		Query: "SELECT i FROM mytable WHERE s = 'first row' ORDER BY i DESC LIMIT ?;",
+		Bindings: map[string]sql.Expression{
+			"v1": expression.NewLiteral(1, sql.Int8),
+		},
+		Expected: []sql.Row{{int64(1)}},
+	},
+	{
+		Query: "SELECT i FROM mytable ORDER BY i LIMIT ? OFFSET 2;",
```
Comment (Hydrocharged) [inline enginetest/queries.go:1240]: It's okay to have unused bindings?

**Review by Hydrocharged — APPROVED** [review]
LGTM!

File: enginetest/queries.go (hunk 1 above)
Comment (zachmu) [reply enginetest/queries.go:1240]: Apparently? Should probably be a bug, I'll file an issue

Comment (zachmu) [reply enginetest/queries.go:1240]: https://github.com/dolthub/go-mysql-server/issues/446

**Pull request merged**